cliquet/scripts/cliquet.py

```
@@ -15,6 +17,12 @@ def deprecated_init(env):
 
 def init_schema(env):
     registry = env['registry']
+    settings = registry.settings
+
+    if settings['read_only']:
+        message = 'You cannot migrate the database with a read_only setup.'
+        warnings.warn(message, ReadOnlyError)
+        return
```

We may want to return an error code there.